DigitalLearningSolutions.Web/Controllers/NotificationPreferencesController.cs

```diff
@@ -66,28 +71,35 @@ public IActionResult UpdateNotificationPreferences(UserType? userType)
         [Route("/NotificationPreferences/Edit/{userType}")]
         public IActionResult SaveNotificationPreferences(UserType? userType, IEnumerable<int> notificationIds)
         {
-            var userId = GetUserId(userType);
-            if (userId == null)
+            var userReference = GetUserReference(userType);
+            if (userReference == null)
             {
                 return NotFound();
             }
 
-            notificationPreferencesService.SetNotificationPreferencesForUser(userType!, userId, notificationIds);
+            notificationPreferencesService.SetNotificationPreferencesForUser(
+                userReference.UserType,
+                userReference.Id,
+                notificationIds
+            );
 
             return RedirectToAction("Index", "NotificationPreferences");
         }
 
-        private int? GetUserId(UserType? userType)
+        private UserReference? GetUserReference(UserType? userType)
         {
+            int? userId = null;
+
             if (Equals(userType, UserType.AdminUser))
             {
-                return User.GetAdminId();
+                userId = User.GetAdminId();
             }
-            if (Equals(userType, UserType.DelegateUser))
+            else if (Equals(userType, UserType.DelegateUser))
             {
-                return User.GetCandidateId();
+                userId = User.GetCandidateId();
             }
-            return null;
+
+            return userId != null ? new UserReference(userId.Value, userType!) : null;

```

Comment:
This is really minor, but this might be easier to follow if we return immediately when reaching a true condition (overriding variables always makes the logic a bit trickier than immediate return):

```
 if (Equals(userType, UserType.AdminUser))
{
return new UserReference(User.GetAdminId().Value, UserType.AdminUser) 
}
else if (Equals(userType, UserType.DelegateUser))
{
return new UserReference(User.GetCandidateId().Value, UserType.DelegateUser) 
}
return null;
}
```

Reply:
Actually, ignore this - we could have null id even if the usertype is different! 👍 🤦